File: src/Editor/Entity/Color.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Editor
{
    public class Color
    {
        public uint r, g, b, a;

        public Color()
        {

        }

        public Color(uint r, uint g, uint b)
        {
            this.r = r;
            this.g = g;
            this.b = b;
        }

        public static Color Black
        {
            get { return new Color(0, 0, 0); }
        }

        public static Color Green
        {
            get { return new Color(0, 255, 0); }
        }

        public static Color Red
        {
            get { return new Color(255, 0, 0); }
        }
    }
}

File: src/Core/Color.cpp
#include "Color.h"

Color Color::White()
{
	Color c;
	c.r = 1.0f;
	c.g = 1.0f;
	c.b = 1.0f;
	c.a = 1.0f;
	
	return c;
}

Color Color::Black()
{
	Color c;
	c.r = 0.0f;
	c.g = 0.0f;
	c.b = 0.0f;
	c.a = 1.0f;

	return c;
}

Color Color::Red()
{
	Color c;
	c.r = 1.0f;
	c.g = 0.0f;
	c.b = 0.0f;
	c.a = 1.0f;

	return c;
}
